echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/cdevents/CDEventTaskRunStarted.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2023 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.echo.cdevents;
+
+import dev.cdevents.CDEvents;
+import dev.cdevents.events.TaskRunStartedCDEvent;
+import io.cloudevents.CloudEvent;
+import java.net.URI;
+
+public class CDEventTaskRunStarted implements CDEventCreator {
+
+  private String source;
```

Comment:
We can probably move a lot of these to some base `CDEvent` class, since I imagine some of these will always be required for all `CDEventCreator`s.

Reply:
yes, making `CDEventCreator` class as an abstract and moving the common properties to this class